INSTALL.md

```diff
@@ -12,7 +12,7 @@ To obtain a copy of the code clone the current repository running
 
 To compile Channelflow:
 * CMake (version 3.1 or higher)
-* C++ supportive compiler
+* C++11 supportive compiler (GNU compiler version 6.0 or higher, clang compiler version 6.0 or higher)
 * FFTW (version 3 or higher)
 * Eigen3 (version 3 or higher)
 
```

Comment:
Why GNU 6.0? Here:
[https://gcc.gnu.org/projects/cxx-status.html](url)

it is written that

> GCC 4.8.1 was the first feature-complete implementation of the 2011 C++ standard, previously known as C++0x.

Reply:
And Clang >= 3.3: https://clang.llvm.org/cxx_status.html

Reply:
We want to enforce users to use the minimum version used in our CI to avoid problems like https://github.com/epfl-ecps/channelflow-pre-release/issues/10 where a Clang version 3.6 is used and it generates a warning about 
`
flowfield.cpp:246:19: error: argument type 'cfbasics::fieldstate *' doesn't match specified 'MPI' type tag that requires 'int *' [-Werror,-Wtype-safety]
MPI_Bcast(&xzstate_, 1, MPI_INT, 0, cfmpi_->comm_world);
`

Reply:
I understand. But that was not about C++11, right? We could say

C++11 supportive compiler (tested with GNU compiler version 4.8.1 or higher, clang compiler version 6.0 or higher)

I would prefer not to ask for too high GNU compilers because from my experience HPC clusters often provide GNU <5.

Reply:
Clang 3.6 complains about broadcasting enum as int. It is a warning which is caught because of treating warnings as errors (check issue number 10 from pre-release). For now, this is not solved yet.

Reply:
@reetzelhaft in the same issue @symtor says that the cose is tested for GNU compiler version 4.8.4.

`
Do you have a module for a higher-version C++ compiler? The code is tested on 4.8.4, g++-6, g++-7, g++-8, and clang-6.0. The minimum compiler requirements are not yet listed in the readme; we'll fix that.
`

Reply:
In principle clang 3.3 supports C++11, so we have three options: we can say that the minimum required compiler is
1 - Clang 3.3, but then we have to fix the warning or we can lower the warning level (long run option).
2 - Clang 3.3, but compilation is only tested for clang 6.0 or higher (which is the truth, even in the future we do not test compilation for lower compilers).
3 - Clang 6.0

Reply:
For reference: I said 4.8.4 because that's the version on our desktops, which we all are probably using, and tests don't fail there.

Reply:
+1 on requiring g++-4.8 or better rather than 6.0. That's what's installed on my UNH cluster, and it works fine.